e2e/test/helpers/templates/FaultInjection.cs

```diff
@@ -76,30 +77,31 @@ public static bool FaultShouldDisconnect(string faultType)
         // --------------------------------------------------------------------------------------------------------------------------------------------------------------------
         //  --- device in normal operation --- | FaultRequested | --- <delayInSec> --- | --- Device in fault mode for <durationInSec> --- | --- device in normal operation ---
         // --------------------------------------------------------------------------------------------------------------------------------------------------------------------
-        public static async Task ActivateFaultInjectionAsync(Client.TransportType transport, string faultType, string reason, TimeSpan delayInSec, TimeSpan durationInSec, DeviceClient deviceClient, MsTestLogger logger)
+        public static async Task ActivateFaultInjectionAsync(Client.TransportType transport, string faultType, string reason, TimeSpan delay, TimeSpan duration, DeviceClient deviceClient, MsTestLogger logger)
         {
-            logger.Trace($"{nameof(ActivateFaultInjectionAsync)}: Requesting fault injection type={faultType} reason={reason}, delay={delayInSec}s, duration={DefaultFaultDuration}s");
-
-            uint oldTimeout = deviceClient.OperationTimeoutInMilliseconds;
+            logger.Trace($"{nameof(ActivateFaultInjectionAsync)}: Requesting fault injection type={faultType} reason={reason}, delay={delay}s, duration={DefaultFaultDuration}s");
 
             try
             {
+                var injectionDelay = TimeSpan.FromMinutes(4);
+
                 // For MQTT FaultInjection will terminate the connection prior to a PUBACK
                 // which leads to an infinite loop trying to resend the FaultInjection message.
-                if (transport == Client.TransportType.Mqtt ||
-                    transport == Client.TransportType.Mqtt_Tcp_Only ||
-                    transport == Client.TransportType.Mqtt_WebSocket_Only)
+                if (transport == Client.TransportType.Mqtt
+                    || transport == Client.TransportType.Mqtt_Tcp_Only
+                    || transport == Client.TransportType.Mqtt_WebSocket_Only)
                 {
-                    deviceClient.OperationTimeoutInMilliseconds = (uint)delayInSec.TotalMilliseconds;
+                    injectionDelay = delay;
```

Comment:
For some reason I can't see this comment in file view, just overview. :(

I don't quite get the purpose of this if statement, but it seems we want to treat MQTT differently than other protocols because we retry automatically.

Since the injectionDelay is initialized to 4 minutes (the old default), this assignment is specifically to MQTT only.

Do you think this makes sense to do?